sdks/java/core/src/main/java/org/apache/beam/sdk/io/Write.java

```diff
@@ -205,13 +249,18 @@ public void populateDisplayData(DisplayData.Builder builder) {
     }
 
     @ProcessElement
-    public void processElement(ProcessContext c) throws Exception {
+    public void processElement(ProcessContext c, BoundedWindow window) throws Exception {
       // Lazily initialize the Writer
       if (writer == null) {
         WriteOperation<T, WriteT> writeOperation = c.sideInput(writeOperationView);
         LOG.info("Opening writer for write operation {}", writeOperation);
         writer = writeOperation.createWriter(c.getPipelineOptions());
-        writer.open(UUID.randomUUID().toString());
+
+        writer.open(UUID.randomUUID().toString(),
```

Comment:
I was expecting to see `writer.openGlobal(base, shardnum, numshards)` and `writer.openWindowed(base, window, pane, shardnum, numshards)`. This way, a writer either supports windowing or not based on the method, not nulls.